src/include/schema.h

```diff
@@ -79,6 +79,27 @@ struct __wt_table {
 #define	WT_COLGROUPS(t)	WT_MAX((t)->ncolgroups, 1)
 
 /*
+ * If a backup cursor is open without schema protection make an effort to
+ * give an error if schema operations will conflict.  This is called after
+ * the schema operations have taken the schema lock so no hot backup
+ * cursor can be created until this is done.
+ */
+#define	WT_SCHEMA_BACKUP_CHECK(session, uri) do {			\
+	__wt_readlock(session, S2C(session)->hot_backup_lock);		\
+	if (S2C(session)->hot_backup) {					\
+		bool in_backup;						\
+		int __ret;						\
+		if ((__ret = __wt_metadata_walk_backup(session,		\
+		    uri, &in_backup)) == 0 && in_backup) {		\
+			__wt_readunlock(session,			\
+			    S2C(session)->hot_backup_lock);		\
+			return (EBUSY);					\
+		}							\
+	} else								\
+		__wt_readunlock(session,				\
+		    S2C(session)->hot_backup_lock);			\
+} while (0)
+/*
```

Comment:
* Why not merge this #define into (a possibly renamed) `__wt_metadata_walk_backup()` function? I can't think of a reason this needs to be a #define.

* The value of `__ret` is being ignored, I think this needs a `WT_RET`.

* Is there any reason to acquire the read-lock before checking the `hot_backup` value? It seems to me that if `hot_backup` is 0, we have no reason to do the lock/unlock, and that's likely to be true most of the time.